### PR TITLE
Artk 139

### PR DESCRIPTION
**Pull request by ThorstenBux**
Packaging for DEB and RPM

### CODE REVIEW

File: share/cmake/modules/FindGStreamer.cmake [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+# The minimum GStreamer version we support.
+SET(GSTREAMER_MINIMUM_VERSION 0.10.30)
+
+# Helper macro to find a GStreamer plugin (or GStreamer itself)
+#   _component_prefix is prepended to the _INCLUDE_DIRS and _LIBRARIES variables (eg. "GSTREAMER_AUDIO")
+#   _pkgconfig_name is the component's pkg-config name (eg. "gstreamer-0.10", or "gstreamer-video-0.10").
+#   _header is the component's header, relative to the gstreamer-0.10 directory (eg. "gst/gst.h").
+#   _library is the component's library name (eg. "gstreamer-0.10" or "gstvideo-0.10")
+MACRO(FIND_GSTREAMER_COMPONENT _component_prefix _pkgconfig_name _header _library)
```
Comment (BoiseBaked) [inline share/cmake/modules/FindGStreamer.cmake]: I would avoid using MACRO in lieu of FUNCTION. Here's why: "If you want true CMake variables and/or better CMake scope control you should look at the function command." MACRO pass args are just preprocessed over ${parm} within the macro body. Very strange.

File: share/packaging/ARToolKit5-bin-win32-vs100.iss [hunk 2]
```diff
@@ -1,345 +1,346 @@
...
-Name: "{group}\Open ARToolKit5 directory"; Filename: "{app}";
-Name: "{group}\Open command console in ARToolKit5 binaries directory"; Filename: "{cmd}"; WorkingDir: "{app}\bin"
-Name: "{group}\ARToolKit - Home (www)"; Filename: "http://www.artoolkit.org/"
-Name: "{group}\ARToolKit - Documentation (www)"; Filename: "http://www.artoolkit.org/documentation"
-Name: "{group}\ARToolKit - API Reference"; Filename: "{app}\doc\apiref\masterTOC.html"
-Name: "{group}\{cm:UninstallProgram,ARToolKit}"; Filename: "{uninstallexe}"
-
-[Registry]
-Root: HKLM; Subkey: "SYSTEM\CurrentControlSet\Control\Session Manager\Environment"; ValueType: string; ValueName: "ARTOOLKIT5_ROOT"; ValueData: {app}; Flags: uninsdeletevalue
+[Setup]
```
Comment (BoiseBaked) [inline share/packaging/ARToolKit5-bin-win32-vs100.iss]: I'm thinking this file may not be merge-able, that is, should be treated as binary file. Not sure though.

File: share/packaging/ARToolKit5-bin-win32-vs120.iss [hunk 3]
```diff
@@ -1,387 +1,388 @@
...
-Name: "{group}\Open ARToolKit5 directory"; Filename: "{app}";
-Name: "{group}\Open command console in ARToolKit5 binaries directory"; Filename: "{cmd}"; WorkingDir: "{app}\bin"
-Name: "{group}\ARToolKit - Home (www)"; Filename: "http://www.artoolkit.org/"
-Name: "{group}\ARToolKit - Documentation (www)"; Filename: "http://www.artoolkit.org/documentation"
-Name: "{group}\ARToolKit - API Reference"; Filename: "{app}\doc\apiref\masterTOC.html"
-Name: "{group}\{cm:UninstallProgram,ARToolKit}"; Filename: "{uninstallexe}"
-
-[Registry]
-Root: HKLM; Subkey: "SYSTEM\CurrentControlSet\Control\Session Manager\Environment"; ValueType: string; ValueName: "ARTOOLKIT5_ROOT"; ValueData: {app}; Flags: uninsdeletevalue
+[Setup]
```
Comment (BoiseBaked) [inline share/packaging/ARToolKit5-bin-win32-vs120.iss]: Ditto the file's content may not be merge-able and therefore should be treated as a binary file.

File: share/packaging/ARToolKit5-bin-winrt-vs120.iss [hunk 4]
```diff
@@ -132,7 +132,6 @@ Source: "include\profile.h"; Components: dev; DestDir: "{app}\include"; Flags: i
 Source: "include\thread_sub.h"; Components: dev; DestDir: "{app}\include"; Flags: ignoreversion
```
Comment (BoiseBaked) [inline share/packaging/ARToolKit5-bin-winrt-vs120.iss]: Ditto on if .iss files should be considered as a binary file type.

File: share/packaging/ARToolKit5Android-bin-bom [hunk 5]
```diff
@@ -55,15 +55,15 @@ bin/pthreadVC2.dll
 bin/glut32.dll
```
Comment (BoiseBaked) [inline share/packaging/ARToolKit5Android-bin-bom]: Not sure if *-bom files are merge-able.

File: share/packaging/ARToolKit5iOS-bin-bom [hunk 6]
```diff
@@ -48,15 +48,15 @@ lib/macosx-universal/libopencv_highgui.2.4.6.dylib
 lib/macosx-universal/libopencv_imgproc.2.4.6.dylib
```
Comment (BoiseBaked) [inline share/packaging/ARToolKit5iOS-bin-bom]: Ditto on -bom files being merge-able.

File: share/packaging/cpack/cpackConfiguration.cmake [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+set(BUILD_ARTEFACTS_PREFIX lib)
+
+###General configuration
+
+##Fill the package control file with the needed information
+#Package control file is required and has some important fields that need to be filled.
+SET(CPACK_PACKAGE_VENDOR "ARToolKit.org")
+SET(CPACK_PACKAGE_CONTACT "info@artoolkit.org")
+set(CPACK_PACKAGE_NAME ${CMAKE_PROJECT_NAME})
```
Comment (BoiseBaked) [inline share/packaging/cpack/cpackConfiguration.cmake]: You want to try to stay consistent with the casing, specially with reserved words of whatever system the text operates over.

File: share/packaging/ARToolKit5-bin-win32-vs100.iss (hunk 2 above)
Comment (philip-lamb) [inline share/packaging/ARToolKit5-bin-win32-vs100.iss]: .iss files are text, but need to retain windows (CRLF) line endings.

File: share/packaging/cpack/cpackStartScripts/startPackaging.sh [hunk 8]
```diff
@@ -0,0 +1,75 @@
+mkdir ../packages
```
Comment (BoiseBaked) [inline share/packaging/cpack/cpackStartScripts/startPackaging.sh]: If this is truly a bash script, the first line needs to be: "#!/bin/bash" (without the quotes).

**Comment by ThorstenBux** [comment]
Close that one, as there went stuff in that was not supposed to be in.

File: share/cmake/modules/FindGStreamer.cmake (hunk 1 above)
Comment (ThorstenBux) [inline share/cmake/modules/FindGStreamer.cmake]: That one I just got online from an other git repository. I haven't looked into it to see what it does. I just use it and it does what I wanted it to do. 
Are you fine with keeping it like that?

File: share/packaging/ARToolKit5-bin-win32-vs100.iss (hunk 2 above)
Comment (ThorstenBux) [inline share/packaging/ARToolKit5-bin-win32-vs100.iss]: This is addressed in the added .gitattributes file

File: share/packaging/ARToolKit5-bin-win32-vs120.iss (hunk 3 above)
Comment (ThorstenBux) [inline share/packaging/ARToolKit5-bin-win32-vs120.iss]: Is merge-able but had wrong line endings.

This is addressed in the added .gitattributes file

File: share/packaging/ARToolKit5-bin-winrt-vs120.iss (hunk 4 above)
Comment (ThorstenBux) [inline share/packaging/ARToolKit5-bin-winrt-vs120.iss]: Is merge-able but had wrong line endings.

This is addressed in the added .gitattributes file

File: share/packaging/ARToolKit5Android-bin-bom (hunk 5 above)
Comment (ThorstenBux) [inline share/packaging/ARToolKit5Android-bin-bom]: they are

File: share/packaging/ARToolKit5iOS-bin-bom (hunk 6 above)
Comment (ThorstenBux) [inline share/packaging/ARToolKit5iOS-bin-bom]: they are

File: share/packaging/cpack/cpackConfiguration.cmake (hunk 7 above)
Comment (ThorstenBux) [inline share/packaging/cpack/cpackConfiguration.cmake]: done

File: share/packaging/cpack/cpackStartScripts/startPackaging.sh (hunk 8 above)
Comment (ThorstenBux) [inline share/packaging/cpack/cpackStartScripts/startPackaging.sh]: I created that for me to make the testing of the cmake packaging stuff easier but as I think it is very handy and useful I committed it. 
I added the "#!/bin/bash" line.

**Pull request closed without merging**